mc-sema/bin_descend/get_cfg.py

```diff
@@ -449,6 +449,54 @@ def isExternalData(fn):
         return False
 
 
+def sanityCheckJumpTableSize(table_ea, ecount):
+    """ IDA doesn't correctly calculate some  jump table sizes """
+    if not isLinkedElf():
+        return ecount
+
+    if getBitness() != 64:
+        return ecount
+
+    table_insn = idautils.DecodeInstruction(table_ea)
+    if table_insn is None:
+        DEBUG("Could not decode instruction at {:x}\n".format(table_insn))
+        return ecount
+
+    if table_insn.Operands[0].type != idc.o_reg:
+        return ecount
+
+    DEBUG("Sanity checking table at {:x}\n".format(table_ea))
+
+    # get register we jump with
+    jmp_reg = table_insn.Operands[0].value
+
+    inst_ea = table_ea
+    # walk back up to 5 instructions
+    for i in xrange(5):
```

Comment:
Is this `cmp` deciding a jump? Should you instead look for the current block head, then at its successor? It seems like the assumption here might be that the jump table is on the fall-through path of the conditional jump (is there one??). I don't think this is an acceptable assumption.